src/gui/inputdevices/PenInputHandler.cpp

```diff
@@ -182,6 +182,11 @@ double PenInputHandler::filterPressure(PositionInputData const& pos, XojPageView
 }
 
 auto PenInputHandler::actionMotion(InputEvent const& event) -> bool {
+    // If we're not handling input, do nothing!
+    if (!this->inputRunning) {
+        return false;
+    }
+
```

Comment:
At present, this PR only changes/adds these 5 lines. As long as this doesn't cause a regression, I would like to keep this PR as small as possible. :)

Reply:
Actually it seems like your fix does prevent the behaviour from happening with other tools(including text tool). Would be great to have this PR merged since it apparently also prevents a destructive action. If you try the same behaviour but with the "Vertical Space" tool, it seems like it actually deletes the elements that it shifts, and they're not recoverable with the "Undo" tool.